phpBB/styles/prosilver/template/viewtopic_body.html

```diff
@@ -224,7 +224,7 @@ <h2 class="poll-title"><!-- EVENT viewtopic_body_poll_question_prepend -->{POLL_
 			<div id="post_content{postrow.POST_ID}"<!-- IF postrow.S_POST_HIDDEN --> style="display: none;"<!-- ENDIF -->>
 
 			<!-- EVENT viewtopic_body_post_subject_before -->
-			<h3 <!-- IF postrow.S_FIRST_ROW -->class="first"<!-- ENDIF -->><!-- IF postrow.POST_ICON_IMG --><img src="{T_ICONS_PATH}{postrow.POST_ICON_IMG}" width="{postrow.POST_ICON_IMG_WIDTH}" height="{postrow.POST_ICON_IMG_HEIGHT}" alt="{postrow.POST_ICON_IMG_ALT}" title="{postrow.POST_ICON_IMG_ALT}" /> <!-- ENDIF --><a href="{postrow.U_MINI_POST}">{postrow.POST_SUBJECT}</a></h3>
+		<h3 {% if postrow.S_FIRST_ROW %}class="first"{% endif %}>{% if postrow.POST_ICON_IMG %}<img src="{{ T_ICONS_PATH }}{{ postrow.POST_ICON_IMG }}" width="{{ postrow.POST_ICON_IMG_WIDTH }}" height="{{ postrow.POST_ICON_IMG_HEIGHT }}" alt="{{ postrow.POST_ICON_IMG_ALT }}" title="{{ postrow.POST_ICON_IMG_ALT }}" /> {% endif %}<a {% if postrow.S_FIRST_ROW && postrow.S_FIRST_UNREAD %}class="first-unread"{% endif %} href="#p{{ postrow.POST_ID }}">{{ postrow.POST_SUBJECT }}</a></h3>
 
```

Comment:
`href="#p{{ postrow.POST_ID }}"` you are replacing the PHP URI which already generates the `#p` fragment correctly hard-coding it? This POST_ID just contains the post identifier  `'POST_ID'			=> $row['post_id'],`

`href="{postrow.U_MINI_POST}"`  I mean already contains the correct fragments plus the POST_ID and moreover applies the `append_sid()` and the correct path:

`'U_MINI_POST'		=> append_sid("{$phpbb_root_path}viewtopic.$phpEx", 'p=' . $row['post_id']) . '#p' . $row['post_id'],`

Reply:
thxs not sure how that got there